code/glTF2Asset.inl

```diff
@@ -187,47 +194,75 @@ inline void LazyDict<T>::DetachFromDocument()
 }
 
 template<class T>
+Ref<T> LazyDict<T>::Retrieve(unsigned int i)
+{
+
+    typename Dict::iterator it = mObjsByOIndex.find(i);
+    if (it != mObjsByOIndex.end()) {// already created?
+        return Ref<T>(mObjs, it->second);
+    }
+
+    // read it from the JSON object
+    if (!mDict) {
+        throw DeadlyImportError("GLTF: Missing section \"" + std::string(mDictId) + "\"");
+    }
+
+    if (!mDict->IsArray()) {
+        throw DeadlyImportError("GLTF: Field is not an array \"" + std::string(mDictId) + "\"");
+    }
+
+    Value& obj = mDict->operator[](i);
```

Comment:
Is this the right way to access the value at `i`? Or is this unsafe/prone-to-failure? I noticed that iterators are used everywhere else (for safety checks?)

Reply:
Function declarations like `operator[]`, `operator+`, etc. are used to overload the standard operators in C++. For example, `std::string` overloads the `+` operator to allow concatenating strings to `char *`s, or to other strings. In this case, rapidjson's `Value` type overloads the `[]` operator, to allow you to do things like `myValue[i]`, instead of `myValue.GetObjectAtIndex(i)`.

So one thing I'd do here is change this line to `Value &obj = (*mDict)[i];`, which is exactly equivalent, but more idiomatic, since it uses the operator directly instead of calling the `operator[]` member function.

That said, you may be right about using iterators instead; it all depends what the `[]` operator actually does. You'll have to look at the rapidjson documentation to see if it performs bounds-checking, etc.

Reply:
I tried implementing it as an iterator, but I couldn't figure out how to do it properly with rapidjson arrays: objects and arrays had different methods for iterating, and returned different values. So I gave up. 🤷‍♂️